app/src/main/java/com/github/epfl/meili/profile/poihistory/PoiHistoryActivity.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.epfl.meili.profile.poihistory
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.github.epfl.meili.R
+import com.github.epfl.meili.database.FirestoreDatabase
+import com.github.epfl.meili.forum.ForumActivity
+import com.github.epfl.meili.home.Auth
+import com.github.epfl.meili.map.MapActivity
+import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.models.VisitedPointOfInterest
+import com.github.epfl.meili.util.MeiliViewModel
+import com.github.epfl.meili.util.TopSpacingItemDecoration
+import java.util.*
+
+
+class PoiHistoryActivity : AppCompatActivity() {
+    companion object {
+        private const val CARD_PADDING: Int = 30
+        private const val ACTIVITY_TITLE = "POI History"
+
+        fun addPoiToHistory(userKey: String, poi: PointOfInterest) {
+            FirestoreDatabase( // add to poi history
+                "poi-history/$userKey/poi-history",
```

Comment:
I think we should try to not hardcode the paths and maybe provide a function that will get you the path given the user key. (c.f. the code review done by the assistants